lib/domain/models/collection.dart

```diff
@@ -64,3 +69,22 @@ class Collection extends MemoExecutionsMetadata with EquatableMixin {
         ...super.props,
       ];
 }
+
+/// Represents all `Collection` metadata
+abstract class CollectionMetadata {
```

Comment:
Why this abstraction?

Reply:
I couldn't find other entity that benefits of it

Reply:
It's used by both `CollectionMemos` and `Collection`, required because we must deal with very similar, but inherently different collection objects, one that is store locally and one that is stored in sembast